ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -92,10 +93,25 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
       ),
     [networkConfigurations],
   );
+
+  // By default, if a non test network is the globally selected network. We will only show non test networks as default selected.
+  const currentlySelectedNetwork = useSelector(getMultichainNetwork);
+  const currentlySelectedNetworkChainId =
+    currentlySelectedNetwork.network.chainId;
+  // If globally selected network is a test network, include that in the default selcted networks for connection request
+  const selectedTestNetwork = testNetworks.filter(
+    (network: { chainId: string }) =>
+      network.chainId === currentlySelectedNetworkChainId,
+  );
+
+  const selectedNetworksList =
+    selectedTestNetwork === undefined
```

Comment:
it's not clear to me how `selectedTestNetwork` would ever be undefined if it's the result of `testNetworks.filter()` which should always return an array

Reply:
I think now since we use `.find`, it makes sense to check for undefined https://github.com/MetaMask/metamask-extension/pull/27980/commits/21f3db86112e79d5c2941ec6f7a94f8179b9fcd1

Reply:
the `selectedNetworksList` assignment can be simplified by leveraging the truthy/falsy nature of `selectedTestNetwork`. Instead of explicitly checking `selectedTestNetwork === undefined`, we can use a direct truthy check:
`
const selectedNetworksList = selectedTestNetwork
  ? [...nonTestNetworks, selectedTestNetwork]
  : nonTestNetworks;
`
this approach achieves the same result, improves readability, and aligns with JavaScript best practices by avoiding redundant === `undefined` checks.

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/27980/commits/f8ee1941029add81790616717f8200a1b57f6c1a